File: ellnix/02/solution.py
from itertools import takewhile

def fibonacci(x = 1, y = 2):
    yield x
    yield from fibonacci(y, x + y)

result = sum(filter(lambda x: x % 2 == 0, takewhile(lambda x: x < 4_000_000, fibonacci())))

print(result)

File: ellnix/02/solution.dart
main() {
  int result = 0;

  for(int x = 1, y = 2, temp = 0; x < 4_000_000; temp = x, x = y, y += temp) {
      if (x % 2 == 0) {
        result += x;
      }
  }

  print(result);
}
